python/lsst/ci/prepare.py

```diff
@@ -151,8 +151,8 @@ def from_product_dict(product_dict):
         Returns:
             The created `Manifest`.
         """
-        deps = [(dep.name, prod.name) for prod in product_dict.values() for dep in prod.dependencies]
-        topo_sorted_product_names = tsort.tsort(deps)
+        deps = [(prod.name, dep.name) for prod in product_dict.values() for dep in prod.dependencies]
+        topo_sorted_product_names = tsort.flatten(tsort.toposort(deps))
```

Comment:
At line 163 below I see the code uses `collections.OrderedDict`. Now that Python `dict` is ordered I strongly suggest changing all such usage to plain `dict`.

Also please consider a dict comprehension for that code below.

Reply:
This code was refactored to simplify the toposort functions and the use of OrderedDict was removed in the code.